Cargo.toml

```diff
@@ -10,6 +18,179 @@ members = [
 	"runtime/*",
 ]
 
+[workspace.dependencies]
+# crates.io
+array-bytes       = { version = "6.0" }
+codec             = { package = "parity-scale-codec", version = "3.3", default-features = false, features = ["derive"] }
+libsecp256k1      = { version = "0.7" }
+scale-info        = { version = "2.3", default-features = false, features = ["derive"] }
+sha3              = { version = "0.9" }
+static_assertions = { version = "1.1" }
+
+# cumulus
+cumulus-pallet-aura-ext               = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-pallet-dmp-queue              = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-pallet-parachain-system       = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-pallet-session-benchmarking   = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-pallet-xcm                    = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-pallet-xcmp-queue             = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-primitives-core               = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-primitives-timestamp          = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-primitives-utility            = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+parachain-info                        = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.36" }
+cumulus-client-cli                    = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
+cumulus-client-collator               = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
+cumulus-client-consensus-aura         = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
+cumulus-client-consensus-common       = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
+cumulus-client-network                = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
+cumulus-client-service                = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
+cumulus-primitives-parachain-inherent = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
+cumulus-relay-chain-interface         = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.36" }
```

Comment:
Maybe we shouldn't put dependencies of external repos  into the workspace, such as cumulus or substrate, etc. 
[Polkadot didn't move them out to workspace.](https://github.com/paritytech/polkadot/blob/master/primitives/Cargo.toml#L14)

Reply:
I see frontier has adopted similar changes recently. This pr is acceptable since it is only about refactoring rust dependencies. I do not touch any logic. The compiler will fire first if something breaks.